Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -98,65 +110,69 @@ public final class RelayClient {
+            .wrapToIridium()
+            .asRPCRequest()
+        let message = try request.asJSONEncodedString()
+        logger.debug("Publishing payload on topic: \(topic)")
+        try await dispatcher.send(message)
     }
 
     /// Completes with an acknowledgement from the relay network.
-    @discardableResult public func publish(
+    public func publish(
         topic: String,
         payload: String,
         tag: Int,
         prompt: Bool = false,
-        onNetworkAcknowledge: @escaping ((Error?) -> Void)) -> Int64 {
-        let params = RelayJSONRPC.PublishParams(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag)
-        let request = JSONRPCRequest<RelayJSONRPC.PublishParams>(method: RelayJSONRPC.Method.publish.method, params: params)
-        let requestJson = try! request.json()
-        logger.debug("iridium: Publishing Payload on Topic: \(topic)")
+        onNetworkAcknowledge: @escaping ((Error?) -> Void)
+    ) {
+        let rpc = Publish(params: .init(topic: topic, message: payload, ttl: defaultTtl, prompt: prompt, tag: tag))
+        let request = rpc
+            .wrapToIridium()
+            .asRPCRequest()
+        let message = try! request.asJSONEncodedString()
+        logger.debug("Publishing Payload on Topic: \(topic)")
         var cancellable: AnyCancellable?
-        dispatcher.send(requestJson) { [weak self] error in
+        cancellable = requestAcknowledgePublisher
+            .filter { $0 == request.id }
+            .sink { (_) in
+            cancellable?.cancel()
+                onNetworkAcknowledge(nil)
+        }
+        dispatcher.send(message) { [weak self] error in
             if let error = error {
                 self?.logger.debug("Failed to Publish Payload, error: \(error)")
                 cancellable?.cancel()
                 onNetworkAcknowledge(error)
             }
         }
-        cancellable = requestAcknowledgePublisher
-            .filter {$0.id == request.id}
-            .sink { (_) in
-            cancellable?.cancel()
-                onNetworkAcknowledge(nil)
-        }
-        return request.id
     }
 
     @available(*, renamed: "subscribe(topic:)")
     public func subscribe(topic: String, completion: @escaping (Error?) -> Void) {
-        logger.debug("iridium: Subscribing on Topic: \(topic)")
-        let params = RelayJSONRPC.SubscribeParams(topic: topic)
-        let request = JSONRPCRequest(method: RelayJSONRPC.Method.subscribe.method, params: params)
-        let requestJson = try! request.json()
+        logger.debug("Relay: Subscribing to topic: \(topic)")
+        let rpc = Subscribe(params: .init(topic: topic))
+        let request = rpc
+            .wrapToIridium()
+            .asRPCRequest()
+        let message = try! request.asJSONEncodedString()
         var cancellable: AnyCancellable?
-        dispatcher.send(requestJson) { [weak self] error in
+        cancellable = subscriptionResponsePublisher
```

Comment:
is it possible to replace completion with withCheckedThrowingContinuation? To make this method async?

Reply:
The problem is that we need to wait for the acknowledgement with the subscription ID, the same problem that I was talking about in out sync meeting, that I couldn't solve yet